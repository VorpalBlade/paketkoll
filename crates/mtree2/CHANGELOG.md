
# 0.4.1

## Added

 - Added more documentation

## Changed

 - Change some potential panics to errors.

## Removed

 - Removed test that didn't test anyting useful.

## New contributors

 - @lucab
