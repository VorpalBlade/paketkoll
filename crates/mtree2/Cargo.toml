[package]
authors = ["Richard Dodd <richard.o.dodd@gmail.com>", "Arvid Norlander"]
categories = ["parsing", "os::unix-apis", "filesystem"]
description = "A crate for iterating through the entries of an mtree record file."
edition = "2021"
license = "Apache-2.0 OR MIT"
name = "mtree2"
readme = "README.md"
repository = "https://github.com/VorpalBlade/paketkoll"
version = "0.6.7"

[dependencies]
bitflags.workspace = true
faster-hex.workspace = true
memchr.workspace = true
smallvec.workspace = true

[lints]
workspace = true
