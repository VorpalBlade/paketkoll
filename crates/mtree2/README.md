# mtree2

This is a fork of [mtree-rs](https://github.com/derekdreery/mtree-rs) fixing some issues and improving performance. Unfortunately the upstream
has been dead apart from one comment, so a fork was neccesary.

The original README is reproduced below:

---

A library for iterating through entries of an mtree.

> *mtree* is a data format used for describing a sequence of files. Their location is record, along with optional extra values like checksums, size, permissions etc.

 For details on the spec see [mtree(5)].

## Examples

```rust
use mtree2::MTree;
use std::time::SystemTime;

// We're going to load data from a string so this example with pass doctest,
// but there's no reason you can't use a file, or any other data source.
let raw_data = "
/set type=file uid=0 gid=0 mode=644
./.BUILDINFO time=1523250074.300237174 size=8602 md5digest=13c0a46c2fb9f18a1a237d4904b6916e \
     sha256digest=db1941d00645bfaab04dd3898ee8b8484874f4880bf03f717adf43a9f30d9b8c
./.PKGINFO time=1523250074.276237110 size=682 md5digest=fdb9ac9040f2e78f3561f27e5b31c815 \
     sha256digest=5d41b48b74d490b7912bdcef6cf7344322c52024c0a06975b64c3ca0b4c452d1
/set mode=755
./usr time=1523250049.905171912 type=dir
./usr/bin time=1523250065.373213293 type=dir
";
let entries = MTree::from_reader(raw_data.as_bytes());
for entry in entries {
    // Normally you'd want to handle any errors
    let entry = entry.unwrap();
    // We can print out a human-readable copy of the entry
    println!("{}", entry);
    // Let's check that if there is a modification time, it's in the past
    if let Some(time) = entry.time() {
        assert!(time < SystemTime::now());
    }
    // We might also want to take a checksum of the file, and compare it to the digests
    // supplied by mtree, but this example doesn't have access to a filesystem.
}
```

[mtree(5)]: https://www.freebsd.org/cgi/man.cgi?mtree(5)
