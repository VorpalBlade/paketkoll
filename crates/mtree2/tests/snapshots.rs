use insta::assert_debug_snapshot;
use mtree2::MTree;
use std::fs::File;
use std::path::PathBuf;

macro_rules! test_snapshot {
    ($name:ident, $path:expr) => {
        #[test]
        fn $name() {
            let path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join($path);
            let mtree = MTree::from_reader(File::open(path).unwrap());
            let entries: Vec<_> = mtree.collect();
            assert_debug_snapshot!(entries);
        }
    };
}

test_snapshot!(test_gedit, "examples/gedit.mtree");
test_snapshot!(test_xterm, "examples/xterm.mtree");
