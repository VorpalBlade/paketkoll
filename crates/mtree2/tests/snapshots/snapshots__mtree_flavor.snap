---
source: crates/mtree2/tests/snapshots.rs
expression: entries
---
[
    Ok(
        Entry {
            path: ".",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 493,
                    },
                ),
                nlink: Some(
                    2,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: None,
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 534794895,
                    },
                ),
                file_type: Some(
                    Directory,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./íþȴɛɜɝɟūǶúÐīŻƩþŽŷǾǻû",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 472401922,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ïɢȽƻſĿƿŻǻǯǅƸóŻÇǿǛææǾ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 469661871,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./óƿÿŶſžɷŶßǿǟƾĻǵǴǻƽ÷ůŽ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 513217652,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./öŞǏǔƣŖŻȷŞǻůɻìɆÿĮǵļúź",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 504699519,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./þſýɝüïžſÿ»ǿŷȳǻÿÿǳĿůý",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 530779925,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ÿȽÿǟżóùūǯïǾŚůɳȿÖŹ½ǫŇ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 525705316,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ĿǡìßɿǯǽïüſÿÚǾşÿãǾûķ·",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 499493724,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ŪÇýįÿŽȿŘǳǾūǱſɣǣ÷żŷůƶ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 523061056,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ƿɼâî÷ſŴýƾɽǜɞŝɷſǽÿƷĻŦ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 535746522,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǧžÿşĻżŹſſǗǷǮħÞǻîȯōɓï",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 463253122,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǭÿŷŜŽȹɥƭþȿǪÿŷȾǮŽǽŜǿ÷",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 510596822,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǯŖɳǥǽŶķįǿúǺſîûɯůĪÿſÿ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 528160319,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǶĻÿŗɫŭǿǫɾįĿȿǷɏņɾƷſźſ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 466555735,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./Ƿǿįýǿŝɷë÷Îÿɭɽŧį¢ÿǷéǿ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 533514646,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǻȼÏǟɷħǮȿǦŪǻžſ÷ɻǞƿŻȶǟ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 459967419,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǿÿǍǿĿǾßǿüƿÿǛÿǶǮǿŽǻŏç",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 507827475,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ǿɳŻýǯĪżşūǳǬ¾ŹžëɫǞźßŽ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 502326657,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ȾǻǼɷǆǧçɿ¿ɷ¿ſǻǺŏǟǠŊǛǶ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 520626212,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ɊǖľŽɿçĽÿõŻſƿǾûũɫɦȿƮǿ",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 517923597,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
    Ok(
        Entry {
            path: "./ɔ½ǼûŭťǹǶæžǺǞǯǷȷǳɓŧǞž",
            params: Params {
                checksum: None,
                device: None,
                contents: None,
                flags: Some(
                    [
                        117,
                        97,
                        114,
                        99,
                        104,
                    ],
                ),
                gid: Some(
                    1004,
                ),
                gname: None,
                ignore: false,
                inode: None,
                link: None,
                md5: None,
                mode: Some(
                    FileMode {
                        mode: 420,
                    },
                ),
                nlink: Some(
                    1,
                ),
                no_change: false,
                optional: false,
                resident_device: None,
                rmd160: None,
                sha1: None,
                sha256: None,
                sha384: None,
                sha512: None,
                size: Some(
                    10,
                ),
                time: Some(
                    SystemTime {
                        tv_sec: 1746539656,
                        tv_nsec: 515568834,
                    },
                ),
                file_type: Some(
                    File,
                ),
                uid: Some(
                    1004,
                ),
                uname: None,
            },
        },
    ),
]
